src/Cech_complex/utilities/cech_persistence.cpp

```diff
@@ -25,41 +26,66 @@
 using Simplex_tree = Gudhi::Simplex_tree<Gudhi::Simplex_tree_options_fast_persistence>;
 using Filtration_value = Simplex_tree::Filtration_value;
 
-using Kernel = CGAL::Epeck_d<CGAL::Dynamic_dimension_tag>;
-using Point = typename Kernel::Point_d;
-using Points_off_reader = Gudhi::Points_off_reader<Point>;
-using Cech_complex = Gudhi::cech_complex::Cech_complex<Kernel, Simplex_tree>;
 using Field_Zp = Gudhi::persistent_cohomology::Field_Zp;
 using Persistent_cohomology = Gudhi::persistent_cohomology::Persistent_cohomology<Simplex_tree, Field_Zp>;
 
-void program_options(int argc, char* argv[], std::string& off_file_points, std::string& filediag,
-                     Filtration_value& max_radius, int& dim_max, int& p, Filtration_value& min_persistence);
+void program_options(int argc, char* argv[], std::string& off_file_points, bool& exact, bool& fast,
+                     std::string& filediag, Filtration_value& max_radius, int& dim_max, int& p,
+                     Filtration_value& min_persistence);
+
+template<class Kernel>
+Simplex_tree create_simplex_tree(const std::string &off_file_points, bool exact_version,
+                                 Filtration_value max_radius, int dim_max) {
+  using Point = typename Kernel::Point_d;
+  using Points_off_reader = Gudhi::Points_off_reader<Point>;
+  using Cech_complex = Gudhi::cech_complex::Cech_complex<Kernel, Simplex_tree>;
+
+  Simplex_tree stree;
+
+  Points_off_reader off_reader(off_file_points);
+  Cech_complex cech_complex_from_file(off_reader.get_point_cloud(), max_radius, exact_version);
+  cech_complex_from_file.create_complex(stree, dim_max);
+
+  return stree;
+}
 
 int main(int argc, char* argv[]) {
   std::string off_file_points;
   std::string filediag;
+  bool exact_version = false;
+  bool fast_version = false;
   Filtration_value max_radius;
   int dim_max;
   int p;
   Filtration_value min_persistence;
 
-  program_options(argc, argv, off_file_points, filediag, max_radius, dim_max, p, min_persistence);
+  program_options(argc, argv, off_file_points, exact_version, fast_version, filediag, max_radius, dim_max, p,
+                  min_persistence);
 
-  Points_off_reader off_reader(off_file_points);
-  Cech_complex cech_complex_from_file(off_reader.get_point_cloud(), max_radius);
+  if ((exact_version) && (fast_version)) {
+    std::cerr << "You cannot set the exact and the fast version." << std::endl;
+    exit(-1);
+  }
 
-  // Construct the Cech complex in a Simplex Tree
-  Simplex_tree simplex_tree;
+  Simplex_tree stree;
+  if (fast_version) {
+    // WARNING : CGAL::Epick_d is fast but not safe (unlike CGAL::Epeck_d)
+    // (i.e. when the points are on a grid)
+    using Fast_kernel = CGAL::Epick_d<CGAL::Dynamic_dimension_tag>;
+    stree = create_simplex_tree<Fast_kernel>(off_file_points, exact_version, max_radius, dim_max);
```

Comment:
It looks a bit strange to use `exact_version` in the fast case, but why not, it documents what this parameter is, and we already ensure that it is false elsewhere.